SA3/qa_web/models.py

```diff
@@ -3,12 +3,52 @@
 
 
 class Post(models.Model):
-    content = models.TextField()
-    owner = models.ForeignKey(User, on_delete=models.SET_NULL, null=True)
+    """
+    Parent meta class that describes a post as a publication on the website with certain content
+    created by a certain owner
+    This will be used for inheritance
+    """
+    content = models.TextField(null=True)
+    owner = models.ForeignKey(User, null= True, on_delete=models.SET_NULL)
     creation_date = models.DateTimeField(auto_now_add=True)
+    upvotes = models.IntegerField(default=0)
+    downvotes = models.IntegerField(default=0)
+    
+    class Meta:
+        abstract = True
 
 class Questions(Post):
-    title = models.CharField(max_length=300)
+    """
+    The question (or thread) initiates the interaction between users' answers,
+    comments, and voting
+    """
+    title = models.CharField(max_length=300, null=True)
+    visits = models.IntegerField(default=0)
+    
     def __str__(self):
         return self.title
+    
+class Answers(Post):
+    """
+    The answer is a proposed solution to a certain question
+    There can be many answers to a question and there is only one correct answer per question
+    """
+    question = models.ForeignKey(Questions, null=True, on_delete=models.CASCADE)
+    correct_answer = models.BooleanField(default = False)
 
+class Comments(Post):
+    """
+    A comment can be posted for an entire question or an answer
+    There can be many comments to questions and answers
+    """
+    question = models.ForeignKey(Questions, null=True, on_delete=models.CASCADE)
+    answer = models.ForeignKey(Answers, null=True, on_delete=models.CASCADE)
+    
+class Tags(models.Model):
```

Comment:
I found a nice lib [here](https://github.com/alex/django-taggit) . It seems easy to manipulate. Maybe we could consider using his wheel. :P